### PR TITLE
prototype(worker-tasks): Create pending task store abstraction

### PR DESCRIPTION
**Pull request by enochtangg**

### CODE REVIEW

**Review by wedamija — COMMENTED** [review]
This is too large to review as is. I'd recommend breaking it up, and adding tests to each section as you go

**Comment by john-z-yang** [comment]
> This is too large to review as is. I'd recommend breaking it up, and adding tests to each section as you go

This is not meant to target prod, it's for prototyping only. Apologies for the confusion!

File: src/sentry/taskworker/pending_tasks.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    PENDING = "pending"
+    PROCESSING = "processing"
+    COMPLETE = "complete"
+    FAILURE = "failure"
+    RETRY = "retry"
+
+
+@dataclass
+class PendingTask:
+    proto_task: PendingTaskProto
```
Comment (markstory) [inline src/sentry/taskworker/pending_tasks.py:19]: :+1: Decorating with a 'domain object' is a good path to take. Copying all the values out of the protobuf will be expensive in aggregate.

We might want to add methods/properties to this object that proxy to the `proto_task` and have the `proto_task` be a private property.

File: src/sentry/taskworker/pending_tasks.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+from dataclasses import dataclass
+from enum import Enum
+
+from sentry_protos.hackweek_team_no_celery_pls.v1alpha.pending_task_pb2 import (
```
Comment (markstory) [inline src/sentry/taskworker/pending_tasks.py:4]: We'll have to come up with a better name for this :smile:

File: src/sentry/taskworker/processors/strategy_factory.py [hunk 3]
```diff
@@ -224,3 +120,4 @@ def do_upkeep(
 
     def shutdown(self):
         self.pool.close()
+        self.pool.close()
```
Comment (markstory) [inline src/sentry/taskworker/processors/strategy_factory.py:123]: Why twice?

Comment (john-z-yang) [reply src/sentry/taskworker/processors/strategy_factory.py]: This is a mistake, it's fixed now

**Pull request merged**